packages/cli/src/lib/implementation/config-middleware.ts

```diff
@@ -1,29 +1,36 @@
 import { readCodePushupConfig } from '@code-pushup/core';
-import { CoreConfig } from '@code-pushup/models';
+import { CoreConfig, OnlyPluginsOptions } from '@code-pushup/models';
 import { GeneralCliOptions } from './model';
 
 export async function configMiddleware<
-  T extends Partial<GeneralCliOptions & CoreConfig>,
+  T extends Partial<GeneralCliOptions & CoreConfig & OnlyPluginsOptions>,
 >(processArgs: T) {
   const args = processArgs as T;
   const { config, ...cliOptions } = args as GeneralCliOptions &
-    Required<CoreConfig>;
+    Required<CoreConfig> &
+    OnlyPluginsOptions;
   const importedRc = await readCodePushupConfig(config);
-  const parsedProcessArgs: CoreConfig & GeneralCliOptions = {
-    config,
-    progress: cliOptions.progress,
-    verbose: cliOptions.verbose,
-    upload: {
-      ...importedRc?.upload,
-      ...cliOptions?.upload,
-    },
-    persist: {
-      ...importedRc.persist,
-      ...cliOptions?.persist,
-    },
-    plugins: importedRc.plugins,
-    categories: importedRc.categories,
-  };
+  const parsedProcessArgs: CoreConfig & GeneralCliOptions & OnlyPluginsOptions =
+    {
+      config,
+      progress: cliOptions.progress,
+      verbose: cliOptions.verbose,
+      upload: {
+        ...importedRc?.upload,
+        ...cliOptions?.upload,
+      },
+      persist: {
+        ...importedRc.persist,
+        ...cliOptions?.persist,
+      },
+      plugins: cliOptions.onlyPlugins?.length
+        ? importedRc.plugins.filter(plugin =>
+            cliOptions.onlyPlugins.includes(plugin.slug),
+          )
+        : importedRc.plugins,
```

Comment:
How should we handle when a user passes an invalid slug to `--onlyPlugin`? This code will just ignore it, but maybe better to at least print a warning for the user, so they realize that they maybe just made a typo :thinking: 

Something like:
```
The --onlyPlugin argument references a plugin with "ligthouse" slug, but no such plugin is present in the configuration. Expected one of the following plugin slugs: "eslint", "lighthouse".
```